traffic_portal/test/integration/PageObjects/TopologiesPage.po.ts

```diff
@@ -0,0 +1,101 @@
+    private btnDelete = element(by.xpath("//button[text()='Delete']"));
+    private txtConfirmName = element(by.name('confirmWithNameInput'));
+    private config = require('../config');
+    private randomize = this.config.randomize;
+    async OpenTopologiesPage(){
+        let snp = new SideNavigationPage();
+        await snp.NavigateToTopologiesPage();
+    }
+    async OpenTopologyMenu(){
+        let snp = new SideNavigationPage();
+        await snp.ClickTopologyMenu();
+    }
+    public async CreateTopologies(topologies: Topologies): Promise<boolean> {
+        let basePage = new BasePage();
+        let snp = new SideNavigationPage();
+        await snp.NavigateToTopologiesPage();
+        //click '+'
+        await this.btnCreateNewTopologies.click();
+        await this.txtName.sendKeys(topologies.Name + this.randomize)
+        await this.txtDescription.sendKeys(topologies.DescriptionData + this.randomize)
+        //click add cache group +
+        await this.btnAddCacheGroup.click();
+        //choose type
+        await this.txtCacheGroupType.sendKeys(topologies.Type);
+        await basePage.ClickSubmit();
+        //choose Cachegroup
+        await this.txtSearchCacheGroup.sendKeys(topologies.CacheGroup + this.randomize)
+        if(await browser.isElementPresent(by.xpath("//td[@data-search='^" + topologies.CacheGroup + this.randomize + "$']")) === true){
+            await element(by.xpath("//td[@data-search='^" + topologies.CacheGroup + this.randomize + "$']")).click();
+        }
+        await basePage.ClickSubmit();
+        await basePage.ClickCreate();
+        if(topologies.description === "create a Topologies with empty cachegroup (no server)"){
+            topologies.validationMessage = topologies.validationMessage + this.randomize;
+        }
+        return await basePage.GetOutputMessage().then(value => value === topologies.validationMessage);
+    }
+       
+    async SearchTopologies(nameTopologies:string): Promise<boolean>{
+        let name = nameTopologies + this.randomize;
+        let snp = new SideNavigationPage();
+        await snp.NavigateToTopologiesPage();
+        await this.txtSearch.clear();
+        await this.txtSearch.sendKeys(name);
+        if (await browser.isElementPresent(element(by.xpath("//td[@data-search='^" + name + "$']"))) == true) {
+            await element(by.xpath("//td[@data-search='^" + name + "$']")).click();
+            return true;
+        } 
+        return false;
+    }
+    async DeleteTopologies(topologies: DeleteTopologies):Promise<boolean>{
+        let name = topologies.Name + this.randomize;
+        let basePage = new BasePage();
+        await this.btnDelete.click();
+        await this.txtConfirmName.sendKeys(name);
+        await basePage.ClickDeletePermanently();
+        return await basePage.GetOutputMessage().then(value => value === topologies.validationMessage);
```

Comment:
This class extends `BasePage`, so couldn't you do `this.ClickDeletePermanently();` etc. instead of making a new `BasePage` instance?

Reply:
Okie. Fixed.